pages/product.js

```diff
@@ -120,70 +128,97 @@ class ProductPage extends React.PureComponent{
+                onMouseOver={this.changeImage}
+              />
+              <img
+                className="thumbnail_small"
+                src={this.props.Item.imageUrl}
+                onMouseOver={this.changeImage}
+              />
+              <img
+                className="thumbnail_small"
+                src={this.props.Item.imageUrl}
+                onMouseOver={this.changeImage}
+              />
             </div>
           </div>
 
           <div className="description_container">
             <div className="product_info_container">
-              <h1>{router.query.name}</h1>
-              {router.query.dprice ? (
+              <h1>{this.props.Item.name}</h1>
+              {this.props.Item.originalPrice ? (
                 <s>
                   <Rupee />
-                  {router.query.price}
+                  {this.props.Item.originalPrice}
                 </s>
-              ):(
+              ) : (
                 <span>
                   <Rupee />
-                  {router.query.price}
+                  {this.props.Item.originalPrice}
                 </span>
               )}
-              {router.query.dprice && (
+              {this.props.Item.discountedPrice && (
                 <span>
                   <Rupee />
-                  <strong>{router.query.dprice}</strong>
+                  <strong>{this.props.Item.discountedPrice}</strong>
                 </span>
               )}
               <p>Product description here</p>
             </div>
             <div className="quantity_container">
-              <button className="increment_button" onClick={this.increment}>+</button>
+              <button className="increment_button" onClick={this.increment}>
+                +
+              </button>
               <div className="quantity">{this.state.qty}</div>
-              <button disabled={!this.state.qty} className="decrement_button" onClick={this.decrement}>-</button>
+              <button
+                disabled={!this.state.qty}
+                className="decrement_button"
+                onClick={this.decrement}
+              >
+                -
+              </button>
             </div>
             <div className="delivery_container">
               <h4>Delivery Options</h4>
             </div>
             <div className="action_container">
               <div className="save_button">
-                <Bookmark fill="#fff" style={{marginRight: '10px'}} />
+                <Bookmark fill="#fff" style={{ marginRight: '10px' }} />
                 Save
               </div>
               <div className="add_to_cart_button">Add To Cart</div>
             </div>
           </div>
-
         </div>
       </div>
     );
   }
 }
 
+ProductPage.getInitialProps = async function(context) {
+  const { id } = context.query;
+  const res = await fetch(`${server}/api/items/${id}`);
+  var Item = await res.json();
+  Item = Item.data;
```

Comment:
No vars, use let in case of mutations, however in this case you can use destructuring.

```js
const { data: Item } = await fetch(`${server}/api/items/${id}`).then(r => r.json());
```